inc/ec_base.h

```diff
@@ -228,11 +228,74 @@ typedef struct {
+ *            endian when transferred over the network. In general, data 
+ *            transmitted over the network is big endian ("network byte 
+ *            order"), so DPP attributes must be treated as a different case.
+ *            Users of this struct MUST maintain the invariant that the 
+ *            `attr_id` and `length` are little endian. If anything needs to be
+ *            done with these values on the host, use an `ec_attribute_t`
+ *            instead.
+ * 
+ * @warning When doing any network operations involving DPP attributes, use `ec_net_attribute_t`, not `ec_attribute_t`. 
+ * @warning When executing logic on the host related to `attr_id` and `length`, use `ec_attribute_t`, not `ec_net_attribute_t`. 
+ */
 typedef struct {
+    /**
+     * @brief Identifies the type of the DPP attribute. Assumed to be little endian, as described in EC 8.1.
+     */
     uint16_t attr_id;
+    /**
+     * @brief Length of the following fields in the attribute. Assumed to be little endian, as described in EC 8.1.
+     */
     uint16_t length;
+    /**
+     * @brief Attribute-specific information fields. Endianness varies according to the specific DPP attribute type.
+     */
     uint8_t data[0];
-}__attribute__((packed)) ec_attribute_t;
+}__attribute__((packed)) ec_net_attribute_t;
+
+/**
+ * @brief Represents a DPP attribute that has been converted to host byte ordering. Not intended to be sent over the network. 
+ * 
+ * @paragraph This struct is intended to be used in any operations on DPP
+ *            attributes that occur entirely on the host. For example, after an
+ *            `ec_net_attribute_t` is read from a frame received from
+ *            the network, it must be converted to an `ec_attribute_t` to
+ *            ensure that the `attr_id` and `length` use host byte ordering. 
+ * 
+ * @paragraph Note that this struct includes a pointer, 
+ *            `ec_net_attribute_t *original`. This MUST ALWAYS point to the
+ *            `ec_net_attribute_t` instance that was used to derive this
+ *            `ec_attribute_t` instance. `original` is used in pointer
+ *            arithmetic based on its position in a frame. 
+ * 
+ * @note `ec_net_attribute_t *original` points to the `ec_attribute_t` instance this was derived from.
+ * 
+ * @warning When doing any network operations involving DPP attributes, use `ec_net_attribute_t`, not `ec_attribute_t`. 
+ * @warning When executing logic on the host related to `attr_id` and `length`, use `ec_attribute_t`, not `ec_net_attribute_t`. 
+ */
+typedef struct {
+    /**
+     * @brief Identifies the type of the DPP attribute. Assumed to be stored with host byte ordering.
+     */
+    uint16_t attr_id;
+    /**
+     * @brief Length of the following fields in the attribute. Assumed to be stored with host byte ordering.
+     */
+    uint16_t length;
+    /**
+     * @brief Points to the `ec_attribute_t` instance this `ec_attribute_t` instance was derived from.
```

Comment:
Missed in the renaming
```suggestion
     * @brief Points to the `ec_net_attribute_t` instance this `ec_attribute_t` instance was derived from.
```